test/e2e/tests/tokens/nft/erc1155-interaction.spec.js

```diff
@@ -38,33 +38,27 @@ describe('ERC1155 NFTs testdapp interaction', function () {
         await driver.clickElement('#batchMintButton');
 
         // Notification
-        const windowHandles = await driver.waitUntilXWindowHandles(3);
-        const [extension] = windowHandles;
-        await driver.switchToWindowWithTitle(
-          WINDOW_TITLES.Dialog,
-          windowHandles,
-        );
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
         // Confirm Mint
         await driver.waitForSelector({
           css: '.confirm-page-container-summary__action__name',
           text: 'Deposit',
         });
-        await driver.clickElement({ text: 'Confirm', tag: 'button' });
-        await driver.waitUntilXWindowHandles(2);
-        await driver.switchToWindow(extension);
+        await driver.clickElementAndWaitForWindowToClose({
+          text: 'Confirm',
+          tag: 'button',
+        });
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
         await driver.clickElement(
           '[data-testid="account-overview__activity-tab"]',
         );
-        const transactionItem = await driver.waitForSelector({
+        await driver.waitForSelector({
           css: '[data-testid="activity-list-item-action"]',
           text: 'Deposit',
         });
-        assert.equal(
-          await transactionItem.isDisplayed(),
-          true,
-          `transaction item should be displayed in activity tab`,
-        );
```

Comment:
this is an anti-pattern, that causes the race condition, when we do the assert if the element is updated (from pending to confirmed)